sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -46,12 +51,146 @@
+    ----------
+    series: :obj:`pandas.Series`
+           input series
+    idx: :obj:`int`, :obj:`slice` or :obj:`pandas.Series`
+        input index
+
+    Returns
+    -------
+    :class:`pandas.Series` or an element of the underneath type
+            object of :class:`pandas.Series`
+    """
+
+    _func_name = 'Operator getitem().'
+
+    if not isinstance(self, SeriesGetitemAccessorType):
+        return None
+
+    accessor = self.accessor.literal_value
+
+    if accessor == 'iloc':
+        if isinstance(idx, types.SliceType):
+            def hpat_pandas_series_iloc_slice_impl(self, idx):
+                return pandas.Series(self._series._data[idx])
+
+            return hpat_pandas_series_iloc_slice_impl
+
+        if isinstance(idx, (int, types.Integer, types.Integer)):
+            def hpat_pandas_series_iloc_impl(self, idx):
+                return self._series._data[idx]
+
+            return hpat_pandas_series_iloc_impl
+
+    if accessor == 'iat':
+        if isinstance(idx, (int, types.Integer)):
+            def hpat_pandas_series_iat_impl(self, idx):
+                return self._series._data[idx]
+
+            return hpat_pandas_series_iat_impl
+
+        raise TypingError('{} The index must be a Integer. Given: {}'.format(_func_name, idx))
+
+    if accessor == 'loc':
+        # Note: Loc return Series
+        # Note: Index 0 in slice not supported
+        index_is_none = (self.series.index is None or
+                         isinstance(self.series.index, numba.types.misc.NoneType))
+        if isinstance(idx, types.SliceType) and not index_is_none:
+            def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
+                max_slice = sys.maxsize
+                start = -1
+                stop = -1
+                is_monotonic = True
+                inc_or_dec = 0
+                for i in numba.prange(len(self._series._index)):
+                    if self._series._index[i] == idx.start and start == -1:
+                        start = i
+                    if self._series._index[i] == idx.stop and start != -1 and stop == -1:
+                        stop = i
+                inc_or_dec = (numpy.all(numpy.diff(self._series._index) <= 0) or
+                              numpy.all(numpy.diff(self._series._index) >= 0))
+                if inc_or_dec != 1:
+                    is_monotonic = False
+
+                def search(idx, choice):
+                    x_value = max_slice
+                    tmp = 0
+                    for i in numba.prange(len(self._series._index)):
+                        case_start = idx < self._series._index[i]
+                        case_stop = idx > self._series._index[i]
+                        if choice == 0 and case_start or choice == 1 and case_stop:
+                            if (idx < 0 and self._series._index[i] > 0 or idx > 0 and self._series._index[i] < 0):
+                                t = math.fabs(idx) + math.fabs(self._series._index[i])
```

Comment:
It's not clear what you're trying to achieve with this function. Maybe it's because of the lack of the comments, but it might also be that it can be rewritten in a simplier way / moved from here to a separate file as a common utility function for slices.